source/Halibut.Tests/ParallelRequestsFixture.cs

```diff
@@ -15,54 +17,116 @@ namespace Halibut.Tests
 {
     public class ParallelRequestsFixture : BaseTest
     {
+        [Test]
+        [LatestAndPreviousClientAndServiceVersionsTestCases(testPolling: false, testWebSocket: false, testNetworkConditions: false)]
+        public async Task MultipleRequestsCanBeInFlightInParallel(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                .WithHalibutLoggingLevel(LogLevel.Info)
+                .WithStandardServices()
+                .Build(CancellationToken);
+
+            var lockService = clientAndService.CreateClient<ILockService>();
+
+            int threadCount = 64;
+            long threadCompletionCount = 0;
+            var threads = new List<Thread>();
+            var exceptions = new ConcurrentBag<Exception>();
+
+            var lockFile = Path.GetTempFileName();
+            var requestStartedFilePathBase = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N"));
+            var requestStartedFilePaths = new ConcurrentBag<string>();
+            Logger.Information($"Lock file: {lockFile}");
+            Logger.Information($"Request started files: {requestStartedFilePathBase}");
+            
+            for (var i = 0; i < threadCount; i++)
+            {
+                int iteration = i;
+                var thread = new Thread(() =>
+                {
+                    // Gotta handle exceptions when running in a 
+                    // Thread, or you're gonna have a bad time.
+                    try
+                    {
+                        var requestStartedPath = $"{requestStartedFilePathBase}-{iteration}";
+                        requestStartedFilePaths.Add(requestStartedPath);
+                        lockService.WaitForFileToBeDeleted(lockFile, requestStartedPath);
+                        Interlocked.Increment(ref threadCompletionCount);
+                    }
+                    catch (Exception e)
+                    {
+                        exceptions.Add(e);
+                    }
+                });
+                thread.Start();
+                threads.Add(thread);
+            }
+
+            // Wait for all requests to be started
+            while (requestStartedFilePaths.Any(p => !File.Exists(p)))
```

Comment:
You can use `Wait.For()` here, or at least pass a cancellation token to `Task.Delay`